source/sink/sink_impl.cc

```diff
@@ -104,4 +113,30 @@ FileSinkImpl::LoadExecutionResult(absl::string_view execution_id) const {
   return responses;
 }
 
+absl::Status
+InMemorySinkImpl::StoreExecutionResultPiece(const nighthawk::client::ExecutionResponse& response) {
+  absl::Status status = validateKey(response.execution_id(), false);
+  if (!status.ok()) {
+    return status;
+  }
+  auto iterator =
+      map_.insert({response.execution_id(), std::vector<nighthawk::client::ExecutionResponse>()})
+          .first;
+  iterator->second.push_back(response);
+  return absl::Status();
+}
+
+absl::StatusOr<std::vector<nighthawk::client::ExecutionResponse>>
+InMemorySinkImpl::LoadExecutionResult(absl::string_view execution_id) const {
+  absl::Status status = validateKey(execution_id, false);
+  if (!status.ok()) {
+    return status;
+  }
+  auto iterator = map_.find(execution_id);
+  if (iterator != map_.end()) {
+    return (*iterator).second;
+  }
+  return absl::Status(absl::StatusCode::kNotFound, "Not found");
```

Comment:
Can we expand the error message, giving a better idea as to what wasn't found? Seeing an error message like this one might not be very helpful when debugging.